asdf_astropy/converters/transform/compound.py

```diff
@@ -72,11 +72,13 @@ def from_yaml_tree_transform(self, node, tag, ctx):
 
         left = node["forward"][0]
         if not isinstance(left, Model):
-            raise TypeError("Unknown model type '{}'".format(node["forward"][0]._tag))
+            raise TypeError("Unknown left model type '{}'".format(node["forward"][0]._tag))
 
         right = node["forward"][1]
-        if not isinstance(right, Model) and not (oper == "fix_inputs" and isinstance(right, dict)):
-            raise TypeError("Unknown model type '{}'".format(node["forward"][1]._tag))
+        if (oper == "fix_inputs" and not isinstance(right, dict)) or (
```

Comment:
Could you not combine these two into one by using `isinstance(right, (dict, Model))`?

Reply:
I don't think so, the issue is if its `fix_inputs` then it must be a dictionary. If it is not `fix_inputs` then it must be a model. You cannot have a `fix_inputs` with a model. That is why there are two statements with an `or` between them.

Reply:
I'm not sure if this case would pass the validation for an asdf file, but the original error was attempting to protect against these cases except you could pass a `fix_inputs` with two models. Which would cause it to fail in a very strange way. This covers that case now.